docs/modules/ROOT/pages/libraries.adoc

```diff
@@ -836,6 +836,33 @@ DS.Regex.regexPartialMatch(@'e(?P<mid>.*)o', 'hello')
 }
 ------------------------
 
+### `regexScan(string pattern, string input)`
+Finds all matches of the input against the pattern. If there are any matches, returns a JSON object which has the following structure.
+- `string` - the original input string;
+- `captures` - array of arrays of captured subgroups per find, if any;
+- `namedCaptures` - array of maps of named subgroups, if any;
+
+*Example*
+
+.DataSonnet map:
+------------------------
+DS.Regex.regexScan(@'(?P<user>[a-z]*)@(?P<domain>[a-z]*).org', 'modus@datasonnet.org,box@datasonnet.org')
+------------------------
+.Result
+------------------------
+{
```

Comment:
Hmm, interesting. Just noticed how the regex results always include the full original string in the go version (and ours), not the matched strings. Odd choice, as I'd think a pattern would be "write a quick regex for a piece of content, want to see what it matches", but the only way to do that is "make sure you wrap the whole regex in a capture group". We'll leave it for now, since it mirror the previous work.